torch/nn/modules/transformer.py

```diff
@@ -287,13 +288,17 @@ def forward(
 
         # Prevent type refinement
         make_causal = False
-        if mask is not None:
-            if is_causal:
-                raise RuntimeError("specify either mask or is_causal, but not both")
+        if is_causal is None:
+            if mask is not None:
+                sz = mask.size(0)
+                causal_comparison = torch.triu(
+                    torch.ones(sz, sz, device=mask.device) * float('-inf'), diagonal=1
+                ).to(mask.dtype)
+
+                if torch.equal(mask, causal_comparison):
+                    make_causal = True
 
-        if make_causal:
-            is_causal = True
-            mask = None
+        is_causal = make_causal
```

Comment:
Are there tests for `is_causal`? I don't see any `ModuleInput` samples that set it.

Regardless, I don't think this is quite correct. If the caller explicitly sets `is_causal=True`, it will be overwritten here to be `False`.

Reply:
That can only happen if the attn_mask is not causal.  Arguably, that's the right outcome?